run_analysis.py

```diff
@@ -95,20 +213,35 @@ def build_analysis_list(config):  # {{{
 def run_analysis(config, analyses):  # {{{
 
     # run each analysis task
+    lastException = None
     for analysisTask in analyses:
         try:
             analysisTask.run()
-        except Exception as e:
+        except (Exception, BaseException) as e:
             if isinstance(e, KeyboardInterrupt):
                 raise e
             traceback.print_exc(file=sys.stdout)
             print "ERROR: analysis module {} failed during run".format(
                 analysisTask.taskName)
+            lastException = e
+
+        # write out a copy of the configuration to document the run
+        logsDirectory = build_config_full_path(config, 'output',
+                                               'logsSubdirectory')
+        configFileName = '{}/config.{}'.format(logsDirectory,
+                                               analysisTask.taskName)
+        configFile = open(configFileName, 'w')
+        config.write(configFile)
+        configFile.close()
```

Comment:
semi-unrelated comment: should we add a timestamp to this config file or the log file for that matter?